src/common/utils/envConfig.ts

```diff
@@ -1,13 +1,24 @@
-import dotenv from "dotenv";
-import { cleanEnv, host, num, port, str, testOnly } from "envalid";
+import dotenv from 'dotenv';
+import { cleanEnv, host, num, port, str, testOnly } from 'envalid';
 
 dotenv.config();
 
 export const env = cleanEnv(process.env, {
-  NODE_ENV: str({ devDefault: testOnly("test"), choices: ["development", "production", "test"] }),
-  HOST: host({ devDefault: testOnly("localhost") }),
+  NODE_ENV: str({
+    devDefault: testOnly('test'),
+    choices: ['development', 'production', 'test'],
+  }),
+  DEPLOY_ENV: str({
+    choices: ['development', 'staging', 'production'],
+  }),
+  HOST: host({ devDefault: testOnly('localhost') }),
   PORT: port({ devDefault: testOnly(3000) }),
-  CORS_ORIGIN: str({ devDefault: testOnly("http://localhost:3000") }),
+  CORS_ORIGIN: str({ devDefault: testOnly('http://localhost:3000') }),
   COMMON_RATE_LIMIT_MAX_REQUESTS: num({ devDefault: testOnly(1000) }),
   COMMON_RATE_LIMIT_WINDOW_MS: num({ devDefault: testOnly(1000) }),
+  SERVICE_AMAZON_USER_URL: str({ devDefault: testOnly('https://amazon.com') }),
```

Comment:
Abstracted the potentially sensitive values in the originally provided code.  If one of the URLs was not considered sensitive, we could move them to the project's constants.